### PR TITLE
Color scheme generator(Resubmission)

### PR DESCRIPTION
**Pull request by giridhar7632**

### CODE REVIEW

File: workshops/color_scheme_generator/README.md [hunk 1]
```diff
@@ -37,14 +37,15 @@ There are many benefits of using Functional components. They are :
 
 ## Setup
 
-For setting up the project fork the repl [here](https://repl.it/@Giridharhackclu/color-scheme-generator-starter#README.md). The starter repl contains ['create-react-app'](https://github.com/facebook/create-react-app) installed. There are some basic styles applied, you will get cleared along with the workshop.
+Fork the starter repl [here](https://repl.it/@Giridharhackclu/color-scheme-generator-starter#README.md). The starter repl contains ['create-react-app'](https://github.com/facebook/create-react-app) installed. There are some basic styles applied, you will get cleared along the flow.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:40]: Phrasing issue is still here: "you will get cleared along the flow"

Comment (giridhar7632) [reply workshops/color_scheme_generator/README.md]: Sorry

File: workshops/color_scheme_generator/README.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
+---
+name: "Color Scheme Generator"
+description: "Create color schemes using React Hooks"
+author: "@giridhar7632"
+---
+
+# Color Scheme Generator
+
+In this workshop we are going to create a colorful app - 'random-color scheme generator' using React and [React Hooks](https://reactjs.org/docs/hooks-intro.html). It looks something like this.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:9]: weird phrasing, doesn't sound super human, kinda robotic

File: workshops/color_scheme_generator/README.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+
+![Final Result](https://cloud-16p9skn27.vercel.app/0colorscheme.png)
+
+Here's the [Live demo](https://color-scheme-generator.giridharhackclu.repl.co/) and [Source Code](https://repl.it/@Giridharhackclu/Color-scheme-generator#src/index.js)
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:13]: check capitalisation

File: workshops/color_scheme_generator/README.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+## Prerequisites
+
+Before starting this workshop, you should have some knowledge about [React.js](https://reactjs.org) and some [ES6](https://www.w3schools.com/Js/js_es6.asp) features of JavaScript.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:17]:
```suggestion
Before starting this workshop, you should have some knowledge about both [React.js](https://reactjs.org) as well as some of the new [ES6](https://www.w3schools.com/Js/js_es6.asp) features in JavaScript.
```

File: workshops/color_scheme_generator/README.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+React Fundamentals
+ - [JSX](https://reactjs.org/docs/introducing-jsx.html)
+ - [Components and Props](https://reactjs.org/docs/components-and-props.html)
+ - [State and LifeCycle](https://reactjs.org/docs/state-and-lifecycle.html)
+
+If you are familiar with these, let’s get going.
+
+## React Hooks API
+
+[Hooks](https://reactjs.org/docs/hooks-intro.html) are introduced in React 16.8 version. Previously, [Class components](https://reactjs.org/docs/components-and-props.html#function-and-class-components) were the only way to define a component that had its own state, and lifecycle methods. [Functional components](https://reactjs.org/docs/components-and-props.html#function-and-class-components) of React, which are light and more flexible, are limited in functionality. Hooks enable us to use state and other React features, like lifecycle methods and event handlers, without writing a Class component.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:28]: Weird phrasing throughout, check your tenses.

File: workshops/color_scheme_generator/README.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+
+There are many benefits of using Functional components. They are :
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:30]:
```suggestion
There are many benefits of using Functional components. They are:
```

File: workshops/color_scheme_generator/README.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+- Easier to read, test, and debug.
+- Have better performance.
+- More reusable.
+
+> **Note:** Hooks don’t work inside classes. But you can use them instead of writing classes.
+
+## Setup
+
+Fork the starter repl [here](https://repl.it/@Giridharhackclu/color-scheme-generator-starter#README.md). The starter repl contains ['create-react-app'](https://github.com/facebook/create-react-app) installed and some prewritten styles.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:40]: Weirdly phrased towards the end

File: workshops/color_scheme_generator/README.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+Click the run button to start the live-server (it’ll take a moment to get running the first time). After running successfully, kick down the console. If you can see `Hello world!` on the webpage, you are good to go.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:42]: Weirdly phrased again, how do I kick down the console?

File: workshops/color_scheme_generator/README.md [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+    <div className="container">
+      
+    </div>
+  )
+}
+```
+
+Now let's create elements with colors. But what colors? We'll give them random colors as we go. 
+
+Add a [prop](https://reactjs.org/docs/components-and-props.html#props-are-read-only) `colors`, an `array`, which we will get from other component. Then we will loop through the array with [`map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:66]: This kinda assumes I know how to add a prop, which not everyone will know

File: workshops/color_scheme_generator/README.md [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+    </div>
+  )
+}
+```
+
+It gets the `colors` array as a prop from another component and returns an array of `div` elements with a `backgroundColor`. Each element contains the respective `backgroundColor` in form of hex code. If it gets nothing, it renders `null`. Also, a [`key`](https://reactjs.org/docs/lists-and-keys.html#keys) is given to every element for its identity. 
+
+Now we finished writing our `Color.js` component. Don't worry, nothing will be on the webpage without rendering the component. 
+
+Open the `App.js` component. For getting a random color we are using a package [randomcolor](https://randomcolor.lllllllllllllllll.com/), which is a small library of some random colors. It is already installed in the starter template. Let's import `Color.js` and `randomColor` into our `App.js` component. Add these lines of code to the `App` component.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:93]: Check the phrasing in the middle

File: workshops/color_scheme_generator/README.md [hunk 11]
```diff
@@ -0,0 +1,268 @@
...
+        <Color
+          colors={colors}
+        />
+        <button className="btn" onClick={change}>Change!</button>
+    </div>
+  )
+}
+```
+
+That's it!! Now you can see random color-schemes on the webpage. You can change the colors, clicking the button. The `useState()` and `useEffect` are the most used hooks in react. Check the final code [here](https://color-scheme-generator.giridharhackclu.repl.co/).
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:234]: Weird phrasing at the end

File: workshops/color_scheme_generator/README.md [hunk 12]
```diff
@@ -0,0 +1,268 @@
...
+
+![Final Project](https://cloud-hptr7s5bf.vercel.app/0color-schemer.gif)
+
+There are other additional hooks and are rarely used. You can check them here.
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:238]: " You can check them here."

File: workshops/color_scheme_generator/README.md [hunk 13]
```diff
@@ -0,0 +1,268 @@
...
+- Example-1: [Demo](https://Monochrome-color-schemes.giridharhackclu.repl.co/). [Source code](https://repl.it/@Giridharhackclu/Monochrome-color-schemes#src/App.js).<br>
+Monochrome Color Schemes - It generates monochrome color-schemes. Same as the project in this workshop, but with different types of schemes.
+
+- Example-2: [Demo](https://custom-color-schemer.giridharhackclu.repl.co/). [Source code](https://repl.it/@Giridharhackclu/Custom-color-schemer#src/App.js).<br>
+Analogic Color Schemes -   It generates Analogic-complement color-schemes. Same as the project in this workshop, but with a different type of scheme.
+
+- Example-3: [Demo](https://random-color-scheme.giridharhackclu.repl.co). [Source code](https://repl.it/@Giridharhackclu/desired-color-scheme#src/App.js).<br>
+In this example, the [randomcolor](https://randomcolor.lllllllllllllllll.com/) package is not used. Instead created a custom function to generate random colors.
+
+Use the power of React Hooks in your next React project. Hope you enjoy this workshop! :v:
```
Comment (sampoder) [inline workshops/color_scheme_generator/README.md:267]: fyi the emoji won't render on the site

**Review by sampoder — CHANGES_REQUESTED** [review]
Hey! So I still think that this workshop needs some work on it's grammar and phrasing. I've left some comments as some guiders.

Thanks,

Sam 🚀

**Comment by sampoder** [comment]
O

**Pull request closed without merging**